src/Workspaces/Core/Portable/Indentation/IInferredIndentationService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Indentation
+{
+    internal interface IInferredIndentationService : IWorkspaceService
```

Comment:
I opted for a new interface, rather than using `IIndentationService`, for 2 reasons:
1. `IIndentationService` is an `ILanguageService`, and this is not language specific.
2. It would heavily complicate the implementation of the concrete versions, because of the need to override the behavior of one method but not the other.

Reply:
This is fine. If necessary, code could provide an implementation of this workspace service that in turn delegates to a language service.

Reply:
This decision seems very correct to me.